src/main/java/it/gov/pagopa/pu/sil/connector/actualization/client/LegacyActualizationClient.java

```diff
@@ -16,7 +21,26 @@ public LegacyActualizationClient(ActualizationApisHolder actualizationApisHolder
   }
 
   public PagamentoAggiornato actualization(String accessToken, String serviceUrl, Pagamento pagamento) {
-    return actualizationApisHolder.getAmountUpdatesLegacyApi(accessToken, serviceUrl)
-        .attualizzazione(pagamento);
+    try {
+      return actualizationApisHolder.getAmountUpdatesLegacyApi(accessToken, serviceUrl)
+          .attualizzazione(pagamento);
+    } catch (RestClientException e) {
+      throw mapToActualizationException(e);
+    }
+  }
+
+  private ActualizationException mapToActualizationException(RestClientException e) {
+    if (e instanceof HttpStatusCodeException ex) {
+      String responseBody = ex.getResponseBodyAsString();
+      try {
+        ObjectMapper mapper = new ObjectMapper();
+        Error error = mapper.readValue(responseBody, Error.class);
+        return new ActualizationException(error.getCode().getValue(), error.getMessage());
```

Comment:
let's translate openApi documented code values in it.gov.pagopa.pu.sil.dto.generated.PuSilErrorDTO.CodeEnum